### ISSUE TITLE
Ansible 2.0 - error for custom module with pipelining

### ISSUE THREAD

**Issue by davidebelloni** [ansible/ansible#13815]
If I run ansible 2.0 with a custom bash module the execution fail, for example:

```
$ ansible -m rpm_installed -i proxy.aaa.com, all -vvv
Using /Users/XXX/svnprj/ansible_configs/ansible.cfg as config file
<proxy.aaa.com> ESTABLISH SSH CONNECTION FOR USER: None
<proxy.aaa.com> SSH: EXEC ssh -C -q -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/Users/XXX/.ansible/cp/ansible-ssh-%h-%p-%r -tt proxy.aaa.com '( umask 22 && mkdir -p "$( echo $HOME/.ansible/tmp/ansible-tmp-1452613433.69-108011768581484 )" && echo "$( echo $HOME/.ansible/tmp/ansible-tmp-1452613433.69-108011768581484 )" )'
<proxy.aaa.com> PUT /var/folders/65/h1flbwnx69b180990q31262r0000gq/T/tmpGz_Cfc TO /root/.ansible/tmp/ansible-tmp-1452613433.69-108011768581484/rpm_installed
<proxy.aaa.com> SSH: EXEC sftp -b - -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/Users/XXX/.ansible/cp/ansible-ssh-%h-%p-%r '[proxy.aaa.com]'
<proxy.aaa.com> PUT /var/folders/65/h1flbwnx69b180990q31262r0000gq/T/tmpzbJe3O TO /root/.ansible/tmp/ansible-tmp-1452613433.69-108011768581484/args
<proxy.aaa.com> SSH: EXEC sftp -b - -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/Users/XXX/.ansible/cp/ansible-ssh-%h-%p-%r '[proxy.aaa.com]'
<proxy.aaa.com> ESTABLISH SSH CONNECTION FOR USER: None
<proxy.aaa.com> SSH: EXEC ssh -C -q -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/Users/XXX/.ansible/cp/ansible-ssh-%h-%p-%r proxy.aaa.com 'LANG=it_IT.UTF-8 LC_ALL=it_IT.UTF-8 LC_MESSAGES=it_IT.UTF-8 /bin/bash  /root/.ansible/tmp/ansible-tmp-1452613433.69-108011768581484/args; rm -rf "/root/.ansible/tmp/ansible-tmp-1452613433.69-108011768581484/" > /dev/null 2>&1'
proxy.aaa.com | FAILED! => {
    "changed": false, 
    "failed": true, 
    "invocation": {
        "module_name": "rpm_installed"
    }, 
    "module_stderr": "", 
    "module_stdout": "", 
    "msg": "MODULE FAILURE", 
    "parsed": false
}
```

As you can see in the last "SSH: EXEC" there's nothing before '/root/.ansible/tmp/ansible-tmp-1452613433.69-108011768581484/args' (there should be '/root/.ansible/tmp/ansible-tmp-1452613433.69-108011768581484/rpm_installed')
If I run the same task with ansible 1.9 the execution ends without errors.

The problem seems to be something like that :

```
--- a/lib/ansible/plugins/action/__init__.py
+++ b/lib/ansible/plugins/action/__init__.py
@@ -426,9 +426,8 @@ class ActionBase(with_metaclass(ABCMeta, object)):

         if self._connection.has_pipelining and self._play_context.pipelining and not C.DEFAULT_KEEP_REMOTE_FILES:
             in_data = module_data
-        else:
-            if remote_module_path:
-                cmd = remote_module_path
+        if remote_module_path:
+            cmd = remote_module_path
```

Thanks

**Comment by abadger** [comment]
Since we still have to pass the arguments in via a file we probably need to disable pipelining whenever we're using a non-new-style python module.

**Comment by abadger** [comment]
Should be fixed by 1b82de2354b92d2683c372a4fbd4b905fab2f19d . Pushed to devel and the stable-2.0 branches so it should be in 2.0.1 when we release that.
